code/datums/components/carbon_on_life.dm

```diff
@@ -0,0 +1,18 @@
+/datum/component/carbon_on_life
+	dupe_type = COMPONENT_DUPE_ALLOWED
+	var/datum/callback/callback
+
+// Will call the callback everytime the carbon calls Life() and the carbon is still alive
+/datum/component/carbon_on_life/Initialize(datum/callback/callback)
+	var/mob/living/carbon/C = parent
+	if(!istype(C) || !callback) //Something went wrong
+		return COMPONENT_INCOMPATIBLE
+	src.callback = callback
+	RegisterSignal(C, COMSIG_CARBON_LIFE, .proc/Life)
+
+/datum/component/carbon_on_life/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_CARBON_LIFE)
+
+/datum/component/carbon_on_life/proc/Life(mob/living/carbon/C)
+	if(C.stat != DEAD)
+		callback.Invoke(C)
```

Comment:
This is another one I'm a *bit* sketchy on too...what needs to have a component that gets called every `Life` that isn't an organ, genetic mutation, or reagent?

Reply:
Again some context.
Ive used this one to make a new antag that does things on life. Like our current changeling which just has "handle_changeling" as proc which is called for every carbon in the game.

Reply:
This component is not required. I didn't fully understand signals. Will update the PR soon